src/app/member/components/MemberEditPage.tsx

```diff
@@ -0,0 +1,369 @@
+"use client";
+import React from "react";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import { addDays, format } from "date-fns";
+import { User, UserRole } from "@/constant/constant";
+import { onSubmit } from "./memberActions";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectLabel,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { CalendarIcon } from "lucide-react";
+import { Calendar } from "@/components/ui/calendar";
+import FormWrapper from "../../../common/FormWrapper";
+import { Label } from "@/components/ui/label";
+
+const phoneValidator =
+  /^((\+92)?(0092)?(92)?(0)?)(3)([0-9]{2})((-?)|( ?))([0-9]{7})$/gm;
+const cnicValidator =
+  /^(3)([0-9]{1})([0-9]{3})((-?)|( ?))([0-9]{7})((-?)|( ?))([0-9]{1})$/gm;
+
+export const formSchema = z.object({
+  name: z
+    .string({ message: "Name is required" })
+    .min(3, "Name should have at least 3 characters"),
+  phone: z
+    .string({ message: "Phone is required" })
+    .regex(phoneValidator, "phone number is not valid"),
+  date_of_birth: z.coerce
+    .date()
+    .min(
+      new Date("01-01-1990"),
+      "You can not suitable for this job because your are over age"
+    )
+    .max(
+      new Date(),
+      "You can not suitable for this job because your are under age"
+    ),
+  cnic: z.string().regex(cnicValidator, "CNIC is not valid"),
+  // shift: z.nativeEnum(ShiftTiming, {
+  //   errorMap: () => {
+  //     return { message: "Please select your user type" };
+  //   },
+  // }),
+  // address: z
+  //   .string({ message: "Address is required" })
+  //   .min(10, "Name should have at least 10 characters"),
+  role: z.nativeEnum(UserRole, {
+    errorMap: () => {
+      return { message: "Please select your user type" };
+    },
+  }),
+  // ehad_start_date: z
+  //   .date()
+  //   .min(addDays(new Date(), 30), "Minmum Ehad Duration is one Month"),
+});
+
+export const MemberEditPage = () => {
```

Comment:
I'd rename that to `MemberFormBuilder`. Rename file too.

Reply:
fixed